src/modules/ctypes/freebsd/appindicator.jsm

```diff
@@ -0,0 +1,58 @@
+/* -*- Mode: js2; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+
+var EXPORTED_SYMBOLS = [ "appind3" ];
+
+const APPINDICATOR_LIBNAME = "appindicator3";
+const APPINDICATOR_ABIS    = [ 1 ];
+
+const Cu = Components.utils;
+const Cc = Components.classes;
+const Ci = Components.interfaces;
+
+Cu.import("resource://gre/modules/ctypes.jsm");
+Cu.import("resource://firetray/ctypes/ctypes-utils.jsm");
+Cu.import("resource://firetray/ctypes/linux/gobject.jsm");
```

Comment:
On 03/03/2015 14:28, foudfou wrote:
>
> In src/modules/ctypes/freebsd/appindicator.jsm 
> <https://github.com/foudfou/FireTray/pull/167#discussion_r25728389>:
>
> > @@ -0,0 +1,58 @@
> > +/* -*- Mode: js2; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
> > +
> > +var EXPORTED_SYMBOLS = [ "appind3" ];
> > +
> > +const APPINDICATOR_LIBNAME = "appindicator3";
> > +const APPINDICATOR_ABIS    = [ 1 ];
> > +
> > +const Cu = Components.utils;
> > +const Cc = Components.classes;
> > +const Ci = Components.interfaces;
> > +
> > +Cu.import("resource://gre/modules/ctypes.jsm");
> > +Cu.import("resource://firetray/ctypes/ctypes-utils.jsm");
> > +Cu.import("resource://firetray/ctypes/linux/gobject.jsm");
>
> "resource://firetray/ctypes/*linux*/gobject.jsm" oops.
> I'm afraid I misguided you. While rethinking about it, I believe the 
> best approach is to point the 'FreeBSD' runtimeOS to Linux in FireTray 
> handler around 
> https://github.com/foudfou/FireTray/blob/master/src/modules/FiretrayHandler.jsm#L80. 
> Something like:
>
> |     } else if (this.runtimeOS == "freebsd") {
>        this.runtimeOS = "linux";
> |
>
> I don't think we need to create directory links unless something 
> differ btw. freebsd and linux. And changes will be kept minimal.
> Sorry for the wrong advice. Would you be willing to amend the changes ?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/foudfou/FireTray/pull/167/files#r25728389>.
>
I thought of that last night. I wasn't sure if this.runtimeOS was 
writable and had setup as much. I screwed up and cut/copy/pasting and 
was going to give another go tonight. I guess we're sort of on the same 
page here.

THX!